src/main/java/glowredman/txloader/JarHandler.java

```diff
@@ -0,0 +1,71 @@
+package glowredman.txloader;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.tuple.Pair;
+
+class JarHandler {
+
+    static final Map<String, Path> CACHED_CLIENT_JARS = new HashMap<>();
+    static final Map<String, Path> CACHED_SERVER_JARS = new HashMap<>();
+
+    static Path txloaderCache;
+
+    static void indexJars() {
+        String userHome = System.getProperty("user.home");
+        txloaderCache = Paths.get(userHome, "txloader");
+
+        List<Pair<Path, String>> clientLocations = new ArrayList<>();
+        clientLocations.add(Pair.of(txloaderCache, "client.jar"));
+        clientLocations.add(Pair.of(Paths.get(userHome, "AppData", "Roaming", ".minecraft", "versions"), "%s.jar"));
```

Comment:
These paths are only the ones I identified on my machine (Windows). Paths on different system can be added easily, I just don't know them.